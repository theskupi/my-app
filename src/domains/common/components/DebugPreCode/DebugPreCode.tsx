import * as React from "react";
import { FunctionComponent } from "react";

interface DebugPreCodeInterface {
  title: string;
  object: unknown;
}

const DebugPreCode: FunctionComponent<DebugPreCodeInterface> = ({
  title,
  object,
}) => {
  return (
    <>
      <div data-testid="title">{title}: </div>
      <pre
        style={{
          whiteSpace: "pre-wrap",
        }}
      >
        <code data-testid="content">{JSON.stringify(object, null, 2)}</code>
      </pre>
    </>
  );
};

export default DebugPreCode;
