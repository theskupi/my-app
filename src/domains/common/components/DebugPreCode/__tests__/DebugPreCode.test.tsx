import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import DebugPreCode from "../DebugPreCode";

it("should render title and object", () => {
  const title = "header";
  const content = "content";

  const { getByTestId } = render(
    <DebugPreCode title={title} object={content} />
  );

  expect(getByTestId("title")).toHaveTextContent(new RegExp(title));
  expect(getByTestId("content")).toHaveTextContent(new RegExp(content));
});

it("should render object as JSON string", () => {
  const title = "header";
  const content = { content: "string" };

  const { getByTestId } = render(
    <DebugPreCode title={title} object={content} />
  );

  const renderedContent = getByTestId("content").textContent;

  expect(renderedContent).toBeTruthy();
  expect(JSON.parse(renderedContent as string)).toEqual(content);
});

it("should match snapshot", () => {
  const title = "header";
  const content = { content: "string" };

  const tree = renderer
    .create(<DebugPreCode title={title} object={content} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
