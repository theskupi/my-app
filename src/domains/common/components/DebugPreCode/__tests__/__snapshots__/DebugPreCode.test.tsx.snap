// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  <div
    data-testid="title"
  >
    header
    : 
  </div>,
  <pre
    style={
      Object {
        "whiteSpace": "pre-wrap",
      }
    }
  >
    <code
      data-testid="content"
    >
      {
  "content": "string"
}
    </code>
  </pre>,
]
`;
